--- codegen.c
+++ codegen.c
@@ -370,8 +370,22 @@
             generando_funciones = 0;
             
-            fprintf(out, "    .globl main\n");
-            fprintf(out, "main:\n");
-            
-            // Prólogo
-            int frame_size = (obtener_mem_disponible() + 4 + 15) & ~15; // +4 para ra, alineado a 16 bytes
-            fprintf(out, "    addi sp, sp, -%d\n", frame_size);
-            fprintf(out, "    sw ra, %d(sp)\n", frame_size - 4);
+            Simbolo *sym_main = symtab_buscar("main");
+            if (sym_main && sym_main->es_funcion) {
+                fprintf(out, "    .globl _start_global\n");
+                fprintf(out, "_start_global:\n");
+                int frame_size = (obtener_mem_disponible() + 4 + 15) & ~15;
+                fprintf(out, "    addi sp, sp, -%d\n", frame_size);
+                fprintf(out, "    sw ra, %d(sp)\n", frame_size - 4);
+            } else {
+                fprintf(out, "    .globl main\n");
+                fprintf(out, "main:\n");
+                int frame_size = (obtener_mem_disponible() + 4 + 15) & ~15;
+                fprintf(out, "    addi sp, sp, -%d\n", frame_size);
+                fprintf(out, "    sw ra, %d(sp)\n", frame_size - 4);
+            }
